[2025-05-15T09:53:26.914+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: sales_ingestion.convert_xlsx_to_csv manual__2025-05-15T09:53:26.493963+00:00 [queued]>
[2025-05-15T09:53:26.919+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: sales_ingestion.convert_xlsx_to_csv manual__2025-05-15T09:53:26.493963+00:00 [queued]>
[2025-05-15T09:53:26.919+0000] {taskinstance.py:1283} INFO - 
--------------------------------------------------------------------------------
[2025-05-15T09:53:26.919+0000] {taskinstance.py:1284} INFO - Starting attempt 1 of 1
[2025-05-15T09:53:26.919+0000] {taskinstance.py:1285} INFO - 
--------------------------------------------------------------------------------
[2025-05-15T09:53:26.929+0000] {taskinstance.py:1304} INFO - Executing <Task(PythonOperator): convert_xlsx_to_csv> on 2025-05-15 09:53:26.493963+00:00
[2025-05-15T09:53:26.932+0000] {standard_task_runner.py:55} INFO - Started process 218 to run task
[2025-05-15T09:53:26.934+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'sales_ingestion', 'convert_xlsx_to_csv', 'manual__2025-05-15T09:53:26.493963+00:00', '--job-id', '22', '--raw', '--subdir', 'DAGS_FOLDER/upload_csv_to_postgres.py', '--cfg-path', '/tmp/tmprdf8o3dz']
[2025-05-15T09:53:26.935+0000] {standard_task_runner.py:83} INFO - Job 22: Subtask convert_xlsx_to_csv
[2025-05-15T09:53:26.976+0000] {task_command.py:389} INFO - Running <TaskInstance: sales_ingestion.convert_xlsx_to_csv manual__2025-05-15T09:53:26.493963+00:00 [running]> on host 1049baf79f4e
[2025-05-15T09:53:27.021+0000] {taskinstance.py:1513} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=sales_ingestion
AIRFLOW_CTX_TASK_ID=convert_xlsx_to_csv
AIRFLOW_CTX_EXECUTION_DATE=2025-05-15T09:53:26.493963+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-05-15T09:53:26.493963+00:00
[2025-05-15T09:53:27.027+0000] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/upload_csv_to_postgres.py", line 10, in xlsx_to_csv
    df = pd.read_excel(xlsx_file)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/excel/_base.py", line 364, in read_excel
    io = ExcelFile(io, storage_options=storage_options, engine=engine)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/excel/_base.py", line 1233, in __init__
    self._reader = self._engines[engine](self._io, storage_options=storage_options)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/excel/_openpyxl.py", line 521, in __init__
    import_optional_dependency("openpyxl")
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/compat/_optional.py", line 118, in import_optional_dependency
    raise ImportError(msg) from None
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
[2025-05-15T09:53:27.033+0000] {taskinstance.py:1327} INFO - Marking task as FAILED. dag_id=sales_ingestion, task_id=convert_xlsx_to_csv, execution_date=20250515T095326, start_date=20250515T095326, end_date=20250515T095327
[2025-05-15T09:53:27.042+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 22 for task convert_xlsx_to_csv (Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.; 218)
[2025-05-15T09:53:27.066+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2025-05-15T09:53:27.081+0000] {taskinstance.py:2582} INFO - 0 downstream tasks scheduled from follow-on schedule check
